homework2/task1/for_text.py

```diff
@@ -0,0 +1,106 @@
+"""
+Given a file containing text. Complete using only default collections:
+    1) Find 10 longest words consisting from largest amount of unique symbols
+    2) Find rarest symbol for document
+    3) Count every punctuation char
+    4) Count every non ascii char
+    5) Find most common non ascii char for document
+
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str) -> List[str]:
+    ...
+
+
+def get_rarest_char(file_path: str) -> str:
+    ...
+
+
+def count_punctuation_chars(file_path: str) -> int:
+    ...
+
+
+def count_non_ascii_chars(file_path: str) -> int:
+    ...
+
+
+def get_most_common_non_ascii_char(file_path: str) -> str:
+    ...
+"""
+import unicodedata
+from operator import itemgetter
+from typing import List
+
+
+def get_longest_diverse_words(file_path: str, encoding="utf8") -> List[str]:
+    dict_of_unique = {}
+    file_input = open(file_path, encoding=encoding)
+    for line in file_input:
+        line = line.encode().decode("unicode-escape")
+        for el in line.split():
+            if el not in dict_of_unique.keys():
+                dict_of_unique[el] = len(set(el)) + len(el)
+
+    list_of_unique = [(k, dict_of_unique[k]) for k in dict_of_unique]
```

Comment:
Более нативный вариант:

```suggestion
    list_of_unique = list(dict_of_unique.items())
```